src/actions/channels/CHANNEL_PINS_UPDATE.ts

```diff
@@ -14,6 +14,8 @@ export default class CoreAction extends Action {
 		const channel = guild ? guild.channels.get(data.d.channel_id) : this.client.dms.get(data.d.channel_id);
 		if (!channel || !isTextBasedChannel(channel)) return;
 
+		channel.lastPinTimestamp = data.d.last_pin_timestamp ?? null;
```

Comment:
Can `last_pin_timestamp` ever be `undefined`? If so, you should guard the set with the `isSet` util from this library.